polymetis/polymetis/python/polysim/grpc_sim_client.py

```diff
@@ -150,6 +158,25 @@ def run(self, time_horizon=float("inf")):
             t += 1
             spinner.spin()
 
+    def mirror(self, time_horizon=float("inf")):
+        """Start mirror simulation by setting simulation state
+
+        Args:
+            time_horizon: If finite, the number of timesteps to stop the simulation.
+
+        """
+        self.connection.InitRobotClient(self.metadata.get_proto())
+        t = 0
+        spinner = Spinner(self.hz)
+        while t < time_horizon:
+            log_request_time = self.log_interval > 0 and t % self.log_interval == 0
+            robot_state = self.execute_rpc_call(
+                self.connection.GetMirrorRobotState,
+                [polymetis_pb2.Empty()],
+                log_request_time=log_request_time,
+            )
+            self.set_robot_state(robot_state)
+
```

Comment:
You don’t need a `mirror` method in the Sim Client (`grpc_sim_client.py`) as you can directly call `set_robot_state` to mirror, and `run` to simulate.